packages/react-storage/src/components/StorageManager/utils/uploadFile.ts

```diff
@@ -4,39 +4,59 @@ import {
   TransferProgressEvent,
   UploadDataInput,
   UploadDataOutput,
+  UploadDataWithPathInput,
+  UploadDataWithPathOutput,
 } from 'aws-amplify/storage';
 
 export type UploadFileProps = {
   file: File;
   key: string;
-  level: StorageAccessLevel;
+  path: string;
+  level?: StorageAccessLevel;
   progressCallback: (event: TransferProgressEvent) => void;
   errorCallback: (error: string) => void;
   completeCallback: (event: { key: string | undefined }) => void;
 } & Record<string, any>;
 
+type UploadData = (
+  input: UploadDataInput | UploadDataWithPathInput
+) => UploadDataOutput | UploadDataWithPathOutput;
+
 export function uploadFile({
   file,
   key,
-  level = 'private',
```

Comment:
Do we want to change this default behavior?

Reply:
Discussed offline. We should respect the current default